### DISCUSSION TITLE
Is there a function to get rx ry rz rotations of pose  ?

### DISCUSSION THREAD

**Discussion by sancelot** [stack-of-tasks/pinocchio#1962] [Q&A] [answered]
In this program, is there a function being able to return pose rotation of data.oMi for each joint  ?
https://github.com/stack-of-tasks/pinocchio/blob/master/examples/overview-urdf.cpp

**Comment by stephane-caron** [comment, marked as answer]
Follow the type of an object to figure out the functions it provides. Here ``data.oMi`` is an `SE3` object: search for "SE3" in the [documentation](https://gepettoweb.laas.fr/doc/stack-of-tasks/pinocchio/master/doxygen-html/) and it will lead you to [this page](https://gepettoweb.laas.fr/doc/stack-of-tasks/pinocchio/master/doxygen-html/structpinocchio_1_1SE3Base.html). You will find the `rotation` function in the list of public member functions. It will give you a rotation matrix, which you can then convert in the angular representation you prefer (tip: use anything you'd like except Euler angles).

Alternatively you can use Python to explore functions using TAB autocomplete:

![image](https://github.com/stack-of-tasks/pinocchio/assets/1189580/bb1320d2-3a30-40f2-9b91-934e50000459)

In IPython, type `pinocchio.SE3.some_function_name?` to get the documentation of a function.

**Comment by sancelot** [comment]
Fine.
